### PR TITLE
Feature/26429 Convert the timeseries python sdk to Github actions

### PR DESCRIPTION
**Pull request by TorgeirWaagbo**
# Pull Request Template

## Description

Converted the timeseries python sdk to Github actions, similar is in AE SDK.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Ran the retry pytests.

**Test Configuration**:
* Environment: Dev

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding changes to the architecture
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by janny93 — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+[tool.poetry.dependencies]
+python = "^3.10"
+requests = "^2.31.0"
+azure-identity = "^1.15.0"
+requests-mock = "^1.11.0"
+pytest = "^7.4.3"
+azure-core = "^1.29.5"
+poetry = "^1.7.1"
+build = "^1.0.3"
```
Comment (equinor-jmel) [inline pyproject.toml:21]: Why are you forcing new minimum requirements?

**Review by equinor-jmel — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (TorgeirWaagbo) [reply pyproject.toml:21]: Good question, I see that some of the requirements in the original setup.py file had minimum requirements while some didn't, so should I only have set the minimum requirements on those that also had in the original setup.py file?

Comment (equinor-jmel) [reply pyproject.toml:21]: Yeah, try to use the same minimum requirements so an update of our SDK does not break for customers, unless one of the packages have a SNYK issue

Comment (TorgeirWaagbo) [reply pyproject.toml]: Changed back to use the same minimum requirements except from opentelemetry-instrumentation-requests and cryptography that required newer version to not get SNYK issues.

**Review by equinor-jmel — APPROVED** [review]

**Pull request merged**